### ISSUE TITLE
:four_leaf_clover: `Proposal`: pkg/util Enhancement

### ISSUE THREAD

**Issue by jxs1211** [devstream-io/devstream#833]
## What Would You Like to Add? Why Is This Needed?

After some code investigation, I found there are many plugins depend on the pkg/util, so I suppose it is quite important to make the package more solid, and the way to improve it is various:
- better naming
- better code organization
- refactor to make it easier to test
- ut
- other aspect to make it stronger

## Design
The tools in this directory have been divided into categories, like downloader\gitlab\github\k8s\helm, etc. So we could refactor it one by one, and create some sub task under this proposal,his could help us move faster and more stable.

**Comment by hxcGit** [comment]
@jxs1211  That's right. I think the best way to start this job is to add some unit tests. Care to contribute some PR for it?

**Comment by jxs1211** [comment]
@hxcGit Maybe we could add some sub task to do this
